client/App/pages/LandingPage/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,90 @@
+import styles from './Carousel.module.scss';
+import CarouselItem from './CarouselItem/CarouselItem';
+import { useRef, useEffect, useState } from 'react';
+import CarouselIndicators from './CarouselIndicators/CarouselIndicators';
+import CarouselControls from './CarouselControls/CarouselControls';
+
+interface carouselProps {
+  slides: { image: string; text: string | undefined }[];
+  interval?: number;
+  controls?: boolean;
+  indicators?: boolean;
+  autoPlay?: boolean;
+  children?: JSX.Element;
+}
+
+const Carousel = ({
+  slides,
+  interval = 5000,
+  controls = false,
+  indicators = false,
+  autoPlay = false,
+  children,
+}: carouselProps) => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const slideInterval = useRef<number>();
```

Comment:
Well written logic, I'm impressed. Only some fixes. You don't have to use useRef hook here. You don't have to create ref to window because it's available globally. useRef isn't necessary also use useState hook instead